bundles/asdi/asdi-projection-change/component/card.js

```diff
@@ -21,48 +21,79 @@ function (view, callback) {
     this.errorListing = Oskari.clazz.create('Oskari.projection.change.view.ErrorListing');
     this.create(view);
 }, {
+    eventHandlers: {
+        AfterMapLayerRemoveEvent: function (event) {
+            this.toggleWarningElement();
+        },
+        AfterMapLayerAddEvent: function (event) {
+            this.toggleWarningElement();
+        }
+    },
+    onEvent: function (event) {
+        var handler = this.eventHandlers[event.getName()];
+        if (!handler) {
+            return;
+        }
+        return handler.apply(this, [event]);
+    },
+    getName: function () {
+        return 'Oskari.projection.change.component.card';
+    },
     getElement: function () {
         return this.element;
     },
     setElement: function (el) {
         this.element = el;
     },
-    checkUnsupportedLayers: function () {
+    getImagePath: function ( epsg ) {
+        var name = epsg.replace(':', '');
+        return '/Oskari/bundles/asdi/asdi-projection-change/resources/images/' + name + '.png';
+    },
+    getUnsupportedLayers: function () {
         var me = this;
         var layers = Oskari.getSandbox().getMap().getLayers();
         var unsupportedLayers = layers.filter( function (layer) {
             return !layer.isSupported(me.view.srsName);
         });
-        if ( unsupportedLayers.length !== 0 ) {
-            var warningElement = this.getElement().find('.projection-warning');
-            warningElement.removeClass('oskari-hidden');
-            //warningLink
-            warningElement.on('click', function ( event ) {
-                event.stopPropagation();
-                me.errorListing.show( jQuery(this), unsupportedLayers );
-            });
+        return unsupportedLayers;
+    },
+    toggleWarningElement: function () {
+        if ( this.getUnsupportedLayers().length !== 0 ) {
+            this.getElement().find('.projection-warning').removeClass('oskari-hidden');
```

Comment:
This doesn't toggle. Only removes the class which makes the warning visible. Nothing hides the warning once it's shown.